docs/2-deploying/1-automated-previews.md

```diff
@@ -41,42 +41,52 @@ jobs:
 
 ### Gitlab CI
 
-This job can be pasted into your `.gitlab-ci.yml` at the root of your repository. You are welcome to change the `stage` to whatever fits your needs to allow you to run tests before the preview is generated, and please be sure to assign correct values for the `variables` in the job. Additionally, you'll need to assign values for `ARCHITECT_EMAIL` and `ARCHITECT_PASSWORD` in your repositories CI variables configuration so that `architect login` will complete successfully.
+This job can be pasted into your `.gitlab-ci.yml` at the root of your repository. You are welcome to change the `stage` to whatever fits your needs to allow you to run tests before the preview is generated, and please be sure to assign correct values for the variables in the job. Additionally, you'll need to assign values for variables in the below config not prefixed with `$CI_` in your repository's CI variables configuration so that the architect commands will run successfully.
+
+This configuration takes advantage of GitLab environments in order to give you better control and visibility into what environments exist and what's deployed to them. On PR creation, both a GitLab and Architect environment will be created. The component specified in the repository will be registered with the Architect Cloud and deployed to the environment. When the PR is either merged or closed, the GitLab environment will be automatically deleted and the component deployed to the environment in the Architect Cloud will be destroyed.
 
 ```yaml
-preview:
-  stage: deploy
+# this example assumes that the repo has ARCHITECT_ACCOUNT and ARCHITECT_PLATFORM set as CI/CD variables
+variables:
+  ARCHITECT_ENVIRONMENT: $CI_MERGE_REQUEST_ID
```

Comment:
Commented on the commit by mistake. Did you not like this proposed syntax?
```yaml
variables:
  ARCHITECT_ACCOUNT: my-account
  ARCHITECT_PLATFORM: my-platform
  
deploy_preview:
  stage: preview
  script: |
    architect register architect.yml -t $ARCHITECT_ENVIRONMENT
    architect environment:create || echo 'Environment already exists'
    architect deploy $ARCHITECT_COMPONENT_NAME:$ARCHITECT_ENVIRONMENT --auto_approve
  environment:
    name: architect/$ARCHITECT_ENVIRONMENT
    on_stop: destroy_preview
  variables:
    ARCHITECT_ENVIRONMENT: preview-$CI_MERGE_REQUEST_ID
  rules:
    - if: $CI_MERGE_REQUEST_ID
    
destroy_preview:
  stage: preview
  script: |
    architect destroy -c $ARCHITECT_COMPONENT_NAME:$ARCHITECT_ENVIRONMENT --auto_approve
    architect environment:destroy --auto_approve
  environment:
    name: architect/$ARCHITECT_ENVIRONMENT
    action: stop
  variables:
    ARCHITECT_ENVIRONMENT: preview-$CI_MERGE_REQUEST_ID
  rules:
    - if: $CI_MERGE_REQUEST_ID
      when: manual
```

Reply:
I pushed a slightly updated version of what I'd prefer. We shouldn't need to define those account and platform variables at the top when they're set in secrets. Also, shouldn't we be able to add `$ARCHITECT_ENVIRONMENT` to the global variables for the pipeline?

Reply:
Probably shouldn't add to the global variables since not every job will have $CI_MERGE_REQUEST_ID and you might have other jobs for dev,staging,prod that have their own environment names.

Reply:
@davidthor want to weigh in?